src/Compilers/CSharp/Test/Syntax/Parsing/ParsingErrorRecoveryTests.cs

```diff
@@ -4480,8 +4482,10 @@ public void TestSemicolonAfterAnonymousTypeMemberStart()
             Assert.NotEqual(SyntaxKind.None, ds.Declaration.Variables[0].Initializer.EqualsToken.Kind());
             Assert.NotNull(ds.Declaration.Variables[0].Initializer.Value);
             Assert.Equal(SyntaxKind.AnonymousObjectCreationExpression, ds.Declaration.Variables[0].Initializer.Value.Kind());
-            Assert.Equal(1, file.Errors().Length);
-            Assert.Equal((int)ErrorCode.ERR_RbraceExpected, file.Errors()[0].Code);
+            Assert.Equal(3, file.Errors().Length);
+            Assert.Equal((int)ErrorCode.ERR_SyntaxError, file.Errors()[0].Code);
+            Assert.Equal((int)ErrorCode.ERR_SemicolonExpected, file.Errors()[1].Code);
+            Assert.Equal((int)ErrorCode.ERR_RbraceExpected, file.Errors()[2].Code);
```

Comment:
this looks to have gotten worse. :-/

Reply:
(copied my response from another comment)
> I do consider this expected. Before the ; was interpreted as the end of the initializer expression (as if a } was written). With these changes, if a } or expression follows the semicolon, the ; will not terminate the initializer expression.

> As a result more accurate errors are generated (imo). In this test case for example the 'new' errors are 1. a syntax error as a ; is not expected in an initializer, 2. a missing ; after the next } (which terminates the initializer expression) and 3. the missing } to close the outer-most scope.

I would argue they are more accurate than before, what would you say regarding the above reasoning?